tests/unit/backend/layers/business/test_business.py

```diff
@@ -172,111 +172,146 @@ def test_create_collection_unauthorized_fail(self):
-        fetched_version = self.business_logic.get_collection(version.collection_id, self.user_info)
+        version = self.initialize_published_collection()
+
+        fetched_version = self.business_logic.get_published_collection_version(version.collection_id, self.user_info)
+
         self.assertIsNotNone(fetched_version.published_at)
         self.assertEqual(fetched_version.metadata, version.metadata)
 
-    def test_get_collection_unpublished_fail(self):
+    def test_get_published_collection_version_for_published_collection_is_none(self):
         """
         An unpublished collection should not be obtained using get_collection
         Instead, `get_collection_version` should be used
         """
-        version = self.initialize_private_collection()
-        fetched_version = self.business_logic.get_collection(version.collection_id, self.user_info)
+        version = self.initialize_unpublished_collection()
+        fetched_version = self.business_logic.get_published_collection_version(version.collection_id, self.user_info)
         self.assertIsNone(fetched_version)
 
-    def test_get_collection_version_ok(self):
+    def test_get_collection_version_for_unpublished_collection_ok(self):
         """
-        A collection version can be obtained using `get_collection_version`
+        An unpublished collection version can be obtained using `get_collection_version`
         """
-        version = self.initialize_private_collection()
+        version = self.initialize_unpublished_collection()
+
         fetched_version = self.business_logic.get_collection_version(version.version_id, self.user_info)
+
         self.assertIsNotNone(fetched_version.published_at)
         self.assertEqual(fetched_version.metadata, version.metadata)
 
-class TestGetAllCollections(BaseBusinessLogicTestCase):
+    def test_get_collection_version_for_published_collection_ok(self):
+        """
+        A published collection version can be obtained using `get_collection_version`
+        """
+        version = self.initialize_published_collection()
+
+        fetched_version = self.business_logic.get_collection_version(version.version_id, self.user_info)
+
+        self.assertIsNotNone(fetched_version.published_at)
+        self.assertEqual(fetched_version.metadata, version.metadata)
 
-    def test_setUp(self) -> None:
-        self.collection1 = self.initialize_private_collection(owner="test_user_1")
-        self.collection2 = self.initialize_private_collection(owner="test_user_2")
-        self.collection3 = self.initialize_public_collection(published_at=datetime(2022, 1, 1))
-        self.collection4 = self.initialize_public_collection(published_at=datetime(2021, 1, 1))
-        self.collection5 = self.initialize_public_collection(published_at=datetime(2020, 1, 1))
-        return super().setUp()
 
-    def test_get_all_collections_ok(self):
+class TestGetAllCollections(BaseBusinessLogicTestCase):
+
+    def test_get_all_collections_unfiltered_ok(self):
```

Comment:
I still think this isn't gonna work. The created collections will belong to the lifetime of `TestGetAllCollections` and aren't reset after each test case, so each subsequent test case will have to rely on what's being created by the previous tests, which doesn't make a lot of sense. If we want to do this, we need to split `TestGetAllCollections` into multiple units. Am I wrong?

Reply:
yes, sorry, I meant to comment that every test should re-init the persistence layer.  that should be cheap with our mock persistence layer, right?

Reply:
`BaseBusinessLogicTestCase` could do that in a setup method perhaps

Reply:
I am not sure using a `.reset()` method is a great idea - what about splitting into separate test cases?